### DISCUSSION TITLE
Treat .h files as C++

### DISCUSSION THREAD

**Discussion by rockwotj** [ast-grep/ast-grep#848] [Q&A]
How can I treat `.h` files as C++ I have a project that uses `.h` for C++, how can I configure AST-grep to understand these files are C++ not C? My C++ rule is not matching :)

**Comment by rockwotj** [comment]
Ah I see: https://ast-grep.github.io/reference/sgconfig.html#languageglobs

**Reply by HerringtonDarkholme** [reply]
I think languageGlobs won't work at the moment? Because `.h` is treated as C and it will not be overridden by langaugeGlobs.

It sounds reasonable to make users' langaugeGlobs take precedence over builtin extension.

**Reply by rockwotj** [reply]
But wait - that doesn't seem to work to override languages does it?

**Reply by rockwotj** [reply]
@HerringtonDarkholme yeah that would be awesome to let the languageGlobs take precedence.

**Reply by HerringtonDarkholme** [reply]
Nope, it does not work at the moment.

<img width="907" alt="image" src="https://github.com/ast-grep/ast-grep/assets/2883231/329fae7c-f304-44d5-b73d-f446f14052b4">

**Reply by rockwotj** [reply]
So is the fix as easy as moving the `SupportLang::from_path(..).map(..)` to be after `DynamicLang` and `lang_globs`?

**Reply by HerringtonDarkholme** [reply]
fixed in 6e440a0ff23b3f5eedb718c0504bffba18a768b1

**Reply by rockwotj** [reply]
Awesome thank you! Want is your release cadence just to set expectations?

**Reply by HerringtonDarkholme** [reply]
Released in 0.17.1
